Boolean.CSharp.Main/Core.cs

```diff
@@ -2,13 +2,140 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
+using System.Transactions;
 
 
 namespace Boolean.CSharp.Main
 {
-    public class Core
+    
+    //TransactionClass
+    public class Transaction
     {
+        //Properties
+        public readonly string date;
+        public readonly float ammount;
+        public readonly string transactionType;
+        public readonly string transactionId;
 
+        //Constructor
+        public Transaction(float ammount) 
+        {
+            //Set date, ammount & ID
+            this.date = DateTime.Now.ToString("dd/MM/yyyy");
+            this.ammount = ammount;
+            this.transactionId = Guid.NewGuid().ToString();
 
+            //transactionType is set automatically, in regards to which
+            //amount you want to deposit/withdraw
+            if (ammount >= 0)
+            {
+                this.transactionType = "Deposit";
+            }
+            else 
+            {
+                this.transactionType = "Withdraw";
+            }
+
+        }
+    }
+
+    //BankAccount Class
+    public class BankAccount
+    {
+        //Properties
+        public float currentBalance { get; private set; }
+        public readonly List<float> balanceList;
+        public List<Transaction> transactionsList { get; private set; }
+        public readonly string AccountID;
+
+        //Constructor, by default balance is 0.0,
+        //transactionslists are empty
+        //new GuID is generated
+        public BankAccount()
+        {
+            this.currentBalance = 0.0f;
+            this.balanceList = new List<float>();
+            this.transactionsList = new List<Transaction>();
+            this.AccountID = Guid.NewGuid().ToString();
+        }
+
+        //MakeTransactionfunction
+        public void MakeTransaction(float ammount)
+        {
+            //check if there is enough money, and ammount in transaction is'nt zero
+            //Probably there is much easier way to design this lol
+
+            //Check ammount != 0.0
+            if (ammount != 0.0f) 
+            {
+                //if ammount is negative and you want to make withdraw
+                if (ammount < 0.0f) 
```

Comment:
code seems a bit complex; too many nested control blocks; you can probably achieve it with simpler code (ie `if (amount <= 0) -> error`

Reply:
resolved